clash-lib/src/Clash/Rewrite/WorkFree.hs

```diff
@@ -155,7 +155,7 @@ isWorkFreeClockOrResetOrEnable tcm e =
     case collectArgs e of
```

Comment:
I wonder if this `collectArgs` should also be `collectArgsTicks` where the ticks are ignored...
It's not causing any issues that we know of though, so it's more _thinking out loud_ than requesting changes

Reply:
The API in `castSignal` branch will no longer allows you to drop ticks silently.

Reply:
Good point, I forgot about that. In that case I don't think we need to worry about this (for anyone else reading: `collectArgs` is gone in the cast branch anyway so this call will be changed later)